src/pages/Authentication/Login.jsx

```diff
@@ -0,0 +1,111 @@
+import "./authentication.css";
+import { Link, useNavigate } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { userLogin } from "../../features/authentication/authSlice";
+import { Loader } from "../../Components";
+
+const Login = () => {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [showPassword, setShowPassword] = useState(false);
+  const { encodedToken, loading, userAccountName, errormessage } = useSelector(
+    (store) => store.auth
+  );
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (encodedToken && userAccountName) {
+      navigate("/");
+    }
+    // eslint-disable-next-line
+  }, [encodedToken]);
+
+  const setTestCredentials = (e) => {
+    e.preventDefault();
+    setUsername("maitreyee");
+    setPassword("mait123");
+  };
+
+  const loginHandler = async (e) => {
+    e.preventDefault();
+    if (username && password) {
+      dispatch(userLogin({ username, password }));
+    }
+  };
+
+  return (
+    <div className="auth__background">
+      {loading && <Loader />}
+      {!loading && (
+        <div className="auth__container">
+          <h2 className="auth__container-header logo-name">Photogram</h2>
+          <form>
+            <label htmlFor="username" className="auth__container-label">
+              Username
+            </label>
+            <input
+              type="text"
+              id="username"
+              placeholder="Username"
+              className="auth__container-input"
+              value={username}
+              onChange={(e) => setUsername(e.target.value)}
+              required
+            />
+            <label htmlFor="password" className="auth__container-label">
+              Password
+            </label>
+            <div className="auth__container-passwor auth__container-input">
+              <input
+                type={showPassword ? "text" : "password"}
+                id="password"
+                placeholder="Password"
+                className="auth__container-passinp"
+                value={password}
+                onChange={(e) => setPassword(e.target.value)}
+                required
+              />
+              {showPassword ? (
+                <i
+                  className="fa-solid fa-eye"
+                  onClick={() => setShowPassword(false)}
```

Comment:
Suggestion - Instead of having the onClick event on the icon, you can put it inside a `<span>` and place the onClick on that.

Reply:
Have added the span tag and moved the onclick to the span tag